+sw_tests/+unit_tests/unittest_spinw_addatom.m

```diff
@@ -0,0 +1,185 @@
+        function test_add_atom_with_update_true_different_spin(testCase)
+            pos = [0; 0; 0];
+            testCase.swobj.addatom('r', pos, 'label', 'atom1')
+            testCase.swobj.addatom('r', pos, 'S', 1, 'label', 'atom1')
+            unit_cell = testCase.swobj.unit_cell;
+            testCase.assertEqual(size(unit_cell.r), [3, 1])  % 1 atom
+            testCase.assertEqual(unit_cell.S, 1)  % updated spin
+            testCase.assertEqual(unit_cell.label, {'atom1'})
+        end
+        
+        function test_add_atom_update_false_different_spin(testCase)
+            pos = [0; 0; 0];
+            testCase.swobj.addatom('r', pos, 'label', 'atom1')
+            testCase.verifyWarning(...
+                @() testCase.swobj.addatom('r', pos, 'S', 1, ...
+                    'label', 'atom1', 'update', false), ...
+                'spinw:addatom:WrongInput')  % warns occ > 1
+            unit_cell = testCase.swobj.unit_cell;
+            testCase.assertEqual(size(unit_cell.r), [3, 2])  % 2 atoms
+            testCase.assertEqual(unit_cell.S, [0, 1])
+        end
+        
+        function test_add_atom_will_not_update_different_pos(testCase)
+            pos = [0; 0; 0];
+            testCase.swobj.addatom('r', pos, 'label', 'atom1')
+            testCase.swobj.addatom('r', pos + 0.5, 'S', 1, 'label', 'atom1')
+            unit_cell = testCase.swobj.unit_cell;
+            testCase.assertEqual(size(unit_cell.r), [3, 2])  % 2 atom
+            testCase.assertEqual(unit_cell.S, [0, 1])
+        end
+        
+        function test_add_atom_named_ion_lookup(testCase)
+            testCase.swobj.addatom('r', [0; 0; 0], 'label', 'Mn3+')
+            unit_cell = testCase.swobj.unit_cell;
+            testCase.assertEqual(size(unit_cell.r), [3, 1])
+            testCase.assertEqual(unit_cell.S, 2)
+            testCase.assertEqual(unit_cell.ox, 3)
+            testCase.assertEqual(unit_cell.Z, int32(25))
+            testCase.assertEqual(unit_cell.b(1), -3.73) % coh scat. len
+            testCase.assertEqual(unit_cell.ff(:,1), [0.4198; 6.926972])
+            testCase.assertEqual(unit_cell.label, {'Mn3+'})
+            % check default values
+            for field = {'occ', 'A', 'biso'}
+                testCase.assertEqual(unit_cell.(field{:}), ...
+                    testCase.default_unit_cell.(field{:}))
+            end
+        end
+        
+        function test_add_atom_lookup_by_Z_with_custom_ox(testCase, oxidation)
+            testCase.swobj.addatom('r', [0; 0; 0], 'Z', 26, ...
+                'ox', oxidation)
+            unit_cell = testCase.swobj.unit_cell;
+            testCase.assertEqual(size(unit_cell.r), [3, 1])
+            testCase.assertEqual(unit_cell.S, 0)  % not looked up
+            testCase.assertEqual(unit_cell.ox, oxidation)
+            testCase.assertEqual(unit_cell.Z, int32(26))
+            testCase.assertEqual(unit_cell.label, ...
+                {['Fe' flip(sprintf('%+i', oxidation)), '_1']})
```

Comment:
The `label` could do with being hardcoded otherwise we are sort of repeating the logic in `addatom.m`. You could put it in the `oxidation` input?